doc/http_api.md

```diff
@@ -0,0 +1,108 @@
+# HTTP API
+
+Enables mods to make HTTP requests.
+
+## Settings
+
+To use this API, mods need to be listed in one or the other following 
+settings in CSV format:
+* `secure.trusted_mods`
+* `secure.http_mods`
+
+An example would be:
+
+```
+secure.http_mods = modname1, modnametwo, modenamethree
+```
+
+## Types
+
+### HttpRequest
+
+A table consisting of the following paramaters:
+
+* `url` (required): 
+  * `https://subdomain.domain.tld`
+* `timeout` (optional): 
+  * Time in seconds
+  * Default is what `curl_timeout` setting is set to (NOTE: setting is 
+    in milliseconds)
+* `method` (optional):
+  * Supports `GET`, `POST`, `PUT`, `DELETE`
+  * Default is `GET`
+* `data` (optional):
+  * Supports string or table
+  * Tables are encoded as `x-www-form-urlencoded` key value pairs
+* `post_data` (optional):
+  * Deprecated, see data above
+* `user_agent` (optional):
+  * Replaces the default user agent if provided
+* `extra_headers` (optional):
+  * Table of strings
+  * Each string needs to follow http spec, `"key: value"`
+* `multipart` (optional):
+  * Boolean, default false
+  * Method must be `POST`
+  * Learn more at [MIME (Wikpedia)](https://en.wikipedia.org/wiki/MIME#Multipart_messages)
+
+### HttpRequestResult
+
+A table consisting of the following paramaters
+
+* `completed`:
+  * Boolean
+  * If requested finished
```

Comment:
What precisely does "finished" mean here? The engine docs are clearer about this:

> (either succeeded, failed or timed out)

So basically the purpose of this field is to enable polling.

Reply:
https://github.com/minetest/minetest_docs/pull/56/commits/d8315ffea2abef8748e090468082a81511e0369e